fdbclient/tests/aws_fixture.sh

```diff
@@ -52,34 +52,57 @@ function create_aws_dir {
 # Write out blob_credentials
 # $1 The token to use querying credentials
 # $2 Hostname going to aws
-# $3 The scratch dir to write the blob credentials file into
+# $3 region
+# $4 bucket
+# $5 The scratch dir to write the blob credentials file into
+# $6 Build dir.
 # Echos the blob_credentials file path.
 function write_blob_credentials {
   local token="${1}"
   local local_host="${2}"
-  local dir="${3}"
-  if ! credentials=$( curl -H "X-aws-ec2-metadata-token: ${token}" \
-      http://169.254.169.254/latest/meta-data/iam/security-credentials/foundationdb-dev_node_instance_role ); then
-    echo "ERROR: Failed reading credentials"
-    return 1
-  fi
-  if ! blob_credentials_str=$( echo "${credentials}" | jq --arg host_arg "${local_host}" \
-      '{"accounts": { ($host_arg): {"api_key": .AccessKeyId, "secret": .SecretAccessKey, "token": .Token}}}'); then
-    echo "ERROR: Failed jq'ing ${blob_credentials_str}"
-    return 1
+  local local_region="${3}"
+  local local_bucket="${4}"
+  local dir="${5}"
+  local local_build_dir="${6}"
+  # Is go available? If so, build our little go program and use it writing blob credentials.
+  if command -v go &> /dev/null; then
+    readonly fetcher_dir="${local_build_dir}/packages/docker/fdb-aws-s3-credentials-fetcher"
+    if ! $( cd "${fetcher_dir}"; go build -o fdb-aws-s3-credentials-fetcher *.go ); then
+      echo "ERROR: Failed go compile."
+      return 1
+    fi
+    readonly fetcher="${fetcher_dir}/fdb-aws-s3-credentials-fetcher"
+    if ! $( "${fetcher}"  -dir "${dir}" ); then
+      echo "ERROR: Failed running ${fetcher}."
+      return 1
+    fi
+    # The sript writes a file whose name is s3_blob_credentials.json
+    echo "${dir}/s3_blob_credentials.json"
+  else
+    # No go so fall back to old way of writing blob credentials.
+    if ! credentials=$( curl -H "X-aws-ec2-metadata-token: ${token}" \
+        http://169.254.169.254/latest/meta-data/iam/security-credentials/foundationdb-dev_node_instance_role ); then
```

Comment:
I don't think we want to hardcode any roles here. But I just saw that those changes are only moved around.